src/modules/dashboard/components/ColonyHome/ColonyHome.tsx

```diff
@@ -97,15 +97,23 @@ const ColonyHome = ({ match, location }: Props) => {
    * This problem is made even worse in an production environment where loading
    * times are slow.
    */
-  if (loading || data?.processedColony?.colonyName !== colonyName) {
+  if (
+    loading ||
+    (data?.processedColony && data.processedColony.colonyName !== colonyName)
+  ) {
     return (
       <div className={styles.loadingWrapper}>
         <LoadingTemplate loadingText={MSG.loadingText} />
       </div>
     );
   }
 
-  if (!colonyName || error || !data?.processedColony) {
+  if (
+    !colonyName ||
+    error ||
+    !data?.processedColony ||
+    (data?.colonyAddress as any) instanceof Error
```

Comment:
Why do you need this check for instanceof Error?

It works without it and i think that the error case is covered by `error` check

Reply:
This is correct, it's how we can detect a error in the first resolver we are calling.

See: https://github.com/JoinColony/colonyDapp/blob/a364c74f0d3f7fa356807ad242eea29984e8737b/src/data/resolvers/colony.ts#L226-L243